bundles/org.openhab.binding.modbus.e3dc/src/main/java/org/openhab/binding/modbus/e3dc/internal/dto/InfoBlock.java

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.e3dc.internal.dto;
+
+import java.nio.ByteBuffer;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.openhab.binding.modbus.e3dc.internal.modbus.Data;
+
+/**
+ * The {@link InfoBlock} Data object for E3DC Info Block
+ *
+ * @author Bernd Weymann - Initial contribution
+ */
+@NonNullByDefault
+public class InfoBlock implements Data {
+    private static final StringType EMPTY = new StringType("NULL");
+    public StringType modbusId = EMPTY;
+    public StringType modbusVersion = EMPTY;
+    public DecimalType supportedRegisters = new DecimalType(-1);
+    public StringType manufacturer = EMPTY;
+    public StringType modelName = EMPTY;
+    public StringType serialNumber = EMPTY;
+    public StringType firmware = EMPTY;
+
+    /**
+     * For decoding see Modbus Register Mapping Chapter 3.1.1 page 14
+     *
+     * @param bArray - Modbus Registers as bytes from 40001 to 40067
+     */
+    public InfoBlock(byte[] bArray) {
+        // index handling to calculate the correct start index
+        ByteBuffer wrapper = ByteBuffer.wrap(bArray);
+
+        // first uint16 = 2 bytes - decode magic byte
+        StringBuilder magicByte = new StringBuilder();
+        magicByte.append(String.format("%02X", wrapper.get()));
+        magicByte.append(String.format("%02X", wrapper.get()));
+        this.modbusId = new StringType(magicByte.toString());
```

Comment:
Did you take a look into the class `HexUtils` in the OH core?

Reply:
I checked Hexutils and adapted the conversion.